src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +60,32 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        // Refresh connection is best effort. it should never throw exception
+        private async Task RefreshConnectionAsync(Uri jobServerUrl, VssCredentials jobServerCredential, DelegatingHandler[] delegatingHandler)
+        {
+            Trace.Info($"Refresh JobServer VssConnection to get on a different AFD node.");
+
+            VssConnection newConnection = null;
+
+            try
+            {
+                _hasConnection = false;
+                newConnection = VssUtil.CreateConnection(jobServerUrl, jobServerCredential, delegatingHandler);
+                await newConnection.ConnectAsync();
+                _connection = newConnection;
+            }
+            catch (Exception ex)
```

Comment:
do we need to catch the exception here? we rely on the exception to throw so `ConnectAsync` can keep retry, I think.

Reply:
It's not necessary... I was following the pattern that was in `RunnerServer`, but this catch isn't needed here.